### PR TITLE
feat(vendor-android): fix ConcurrentModificationException in logcat l…

### LINKED ISSUE #310: Exception in thread "LogCatLogger-emulator-5556" java.util.ConcurrentModificationException
I use cli builded from develop https://github.com/Malinskiy/marathon/commit/11fe5894334a46e8af439d36e2eb5f0a933c6ba6

When I run tests I get exception in logs

`
Exception in thread "LogCatLogger-emulator-5556" java.util.ConcurrentModificationException
  at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:909)
  at java.util.ArrayList$Itr.next(ArrayList.java:859)
  at com.malinskiy.marathon.android.ddmlib.DdmlibAndroidDevice$listener$1.invoke(DdmlibAndroidDevice.kt:425)
  at com.malinskiy.marathon.android.ddmlib.DdmlibAndroidDevice$listener$1.invoke(DdmlibAndroidDevice.kt:64)
  at com.malinskiy.marathon.android.ddmlib.DdmlibAndroidDeviceKt$sam$com_android_ddmlib_logcat_LogCatListener$0.log(DdmlibAndroidDevice.kt)
  at com.android.ddmlib.logcat.LogCatReceiverTask.notifyListeners(LogCatReceiverTask.java:128)
  at com.android.ddmlib.logcat.LogCatReceiverTask.access$300(LogCatReceiverTask.java:35)
  at com.android.ddmlib.logcat.LogCatReceiverTask$LogCatOutputReceiver.processLogLines(LogCatReceiverTask.java:113)
  at com.android.ddmlib.logcat.LogCatReceiverTask$LogCatOutputReceiver.processNewLines(LogCatReceiverTask.java:106)
  at com.android.ddmlib.MultiLineReceiver.addOutput(MultiLineReceiver.java:104)
  at com.android.ddmlib.AdbHelper.executeRemoteCommand(AdbHelper.java:572)
  at com.android.ddmlib.AdbHelper.executeRemoteCommand(AdbHelper.java:627)
  at com.android.ddmlib.AdbHelper.executeRemoteCommand(AdbHelper.java:420)
  at com.android.ddmlib.AdbHelper.executeRemoteCommand(AdbHelper.java:347)
  at com.android.ddmlib.Device.executeShellCommand(Device.java:576)
  at com.android.ddmlib.logcat.LogCatReceiverTask.run(LogCatReceiverTask.java:74)
  at com.malinskiy.marathon.android.ddmlib.DdmlibAndroidDevice$logcatThread$1.invoke(DdmlibAndroidDevice.kt:78)
  at com.malinskiy.marathon.android.ddmlib.DdmlibAndroidDevice$logcatThread$1.invoke(DdmlibAndroidDevice.kt:64)
  at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)
`

### PR DESCRIPTION
**Pull request by Malinskiy**
…istener

also update relative version in samples

Fixes #310

### CODE REVIEW

File: vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/line/NullOutputListener.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.malinskiy.marathon.android.executor.listeners.line
+
+/**
+ * Ignores the output
+ */
+class NullOutputListener : LineListener {
+    override fun onLine(line: String) {}
+}
```
Comment (Vacxe) [inline vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/line/NullOutputListener.kt]: CMD + SHIFT + A **->** Ensure line feed at file end on Save option box (5 sec)

File: vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/video/ScreenRecorderTestRunListener.kt [hunk 2]
```diff
@@ -38,16 +38,16 @@ class ScreenRecorderTestRunListener(
 
     private var hasFailed: Boolean = false
     private var recorder: Thread? = null
-    private var receiver: LineListener? = null
+    private var outputListener: LineListener? = null
```
Comment (Vacxe) [inline vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/video/ScreenRecorderTestRunListener.kt]: Maybe make sense to declare with  NullOutputListener() for avoiding `!!` check?

File: vendor/vendor-android/ddmlib/src/main/kotlin/com/malinskiy/marathon/android/ddmlib/DdmlibAndroidDevice.kt [hunk 3]
```diff
@@ -78,21 +78,28 @@ class DdmlibAndroidDevice(
...
+        synchronized(logcatListeners) {
+            logcatListeners.add(listener)
+        }
     }
 
     override fun removeLogcatListener(listener: LineListener) {
-        logcatListeners.getAndUpdate { it.apply { remove(listener) } }
+        synchronized(logcatListeners) {
+            logcatListeners.remove(listener)
+
```
Comment (Vacxe) [inline vendor/vendor-android/ddmlib/src/main/kotlin/com/malinskiy/marathon/android/ddmlib/DdmlibAndroidDevice.kt]: Empty line

File: vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/line/NullOutputListener.kt (hunk 1 above)
Comment (Malinskiy) [reply vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/line/NullOutputListener.kt]: Added .editorconfig to work around this

File: vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/video/ScreenRecorderTestRunListener.kt (hunk 2 above)
Comment (Malinskiy) [reply vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/video/ScreenRecorderTestRunListener.kt]: it's refreshed for every test run, although maybe you're right to reuse this `black hole` :)

**Review by Vacxe — APPROVED** [review]

**Comment by Vacxe** [comment]
Cheers

**Pull request merged**